backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -68,3 +68,17 @@ def create_collection(body: object, user: str):
 
 def get_collection_dataset(dataset_uuid: str):
     raise NotImplementedError
+
+
+def delete_collection(collection_uuid: str, visibility: str, user: str):
+    if visibility == CollectionVisibility.PUBLIC.name:
+        return make_response(jsonify("Can not delete a public dataset"), 405)
```

Comment:
Create an exception for this, inherit ProblemException, and then raise it here. This will ensure the format of the exception is uniform.

Reply:
if we arent including visibility in the api  this will just be removed. If the collection is public it wont be retrieved and the error will be sent from the next lines